dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/AbstractMethodProblem.java

```diff
@@ -17,12 +17,39 @@
 package com.google.cloud.tools.opensource.classpath;
 
 /**
- * The referenced {@code methodSymbol} is not implemented in the {@code targetClass} but the class
- * is declared to implement the method by {@code sourceClass}. Such unimplemented methods manifest
- * as {@link AbstractMethodError}s at runtime.
+ * {@code implementationClass} does not implement the abstract method {@code methodSymbol} declared
+ * by {@code interfaceOrAbstractClass}. Such unimplemented methods manifest as {@link
+ * AbstractMethodError}s at runtime.
  */
 final class AbstractMethodProblem extends LinkageProblem {
-  AbstractMethodProblem(ClassFile sourceClass, ClassFile targetClass, MethodSymbol methodSymbol) {
-    super("is not implemented in the class", sourceClass, methodSymbol, targetClass);
+  MethodSymbol methodSymbol;
+
+  AbstractMethodProblem(
+      ClassFile implementationClass,
+      ClassFile interfaceOrAbstractClass,
+      MethodSymbol methodSymbol) {
```

Comment:
methodSymbol should be the second argument and interfaceOrAbstractClass the third to match the argument order in the superclass constructor.

Reply:
Nice idea. Moved methodSymbol.